### PR TITLE
Feature/initial project structure

### PR DESCRIPTION
**Pull request by Oleg-Mozhey**
We've just launched the initial version of abstractions for UI test cases. While it's not flawless, it marks the beginning of a project (first step) with extensive potential. **If you're interested in contributing and enhancing its structure, your input is more than welcome.**

Background: our BDD project's first version lacked abstractions and was essentially copied as is. This decision hindered the project's reusability and made maintaining the GitHub repository quite challenging. Consequently, we decided to revamp the framework to be more flexible. We believe in making these changes gradually, taking iterative steps forward.

### CODE REVIEW

File: Behavioral.Automation.AsyncAbstractions.UI/StepDefinitions/ButtonStepDefinitions.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using Behavioral.Automation.AsyncAbstractions.UI.Interfaces;
+using TechTalk.SpecFlow;
+
+namespace Behavioral.Automation.AsyncAbstractions.UI.StepDefinitions;
+
+[Binding]
+public class ButtonStepDefinitions
+{
+
+    [Given(@"""(.+?)"" button has been clicked")]
```
[the remark this answers is not in the document]
Comment (Oleg-Mozhey) [reply Behavioral.Automation.AsyncAbstractions.UI/StepDefinitions/ButtonStepDefinitions.cs:10]: I don't mind using the

File: Behavioral.Automation.AsyncAbstractions.UI/Interfaces/IWebElementStorageService.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Behavioral.Automation.AsyncAbstractions.UI.Interfaces;
+
+public interface IWebElementStorageService
```
[the remark this answers is not in the document]
Comment (Oleg-Mozhey) [reply Behavioral.Automation.AsyncAbstractions.UI/Interfaces/IWebElementStorageService.cs:3]: added

File: Behavioral.Automation.AsyncAbstractions.UI/Behavioral.Automation.AsyncAbstractions.UI.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <Nullable>enable</Nullable>
+        <Platforms>AnyCPU</Platforms>
+        <PackageId>Behavioral.Automation.AsyncAbstractions.UI</PackageId>
+        <Version>0.0.1</Version>
+        <Authors>Quantori Inc.</Authors>
+        <Product>tas</Product>
+        <Copyright>Quantori Inc.</Copyright>
+        <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+        <IsPackable>true</IsPackable>
+        <Description>Quantori BDD is a automation framework that makes use of a wide array of existing grammatical structures, including abstractions, implementations, and bindings, while also offering strong customization options for any implementations.
```
[the remark this answers is not in the document]
Comment (Oleg-Mozhey) [reply Behavioral.Automation.AsyncAbstractions.UI/Behavioral.Automation.AsyncAbstractions.UI.csproj:15]: Added

File: Behavioral.Automation.AsyncAbstractions.UI/BasicImplementations/ElementTransformations.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using Behavioral.Automation.AsyncAbstractions.UI.Interfaces;
+using TechTalk.SpecFlow;
+
+namespace Behavioral.Automation.AsyncAbstractions.UI.BasicImplementations;
+
+[Binding]
+public class ElementTransformations
```
[the remark this answers is not in the document]
Comment (Oleg-Mozhey) [reply Behavioral.Automation.AsyncAbstractions.UI/BasicImplementations/ElementTransformations.cs:7]: Added

File: Behavioral.Automation.AsyncAbstractions.UI/BasicImplementations/ElementSelector.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Behavioral.Automation.AsyncAbstractions.UI.BasicImplementations;
+
+public class ElementSelector
```
[the remark this answers is not in the document]
Comment (Oleg-Mozhey) [reply Behavioral.Automation.AsyncAbstractions.UI/BasicImplementations/ElementSelector.cs:3]: Added

**Review by vladimir-s-quantori — APPROVED** [review]

**Pull request merged**